op-interop-mon/monitor/metric_collector_test.go

```diff
@@ -0,0 +1,428 @@
+					blockHash: "0x0000000000000000000000000000000000000000000000000000000000000123",
+					status:    "future",
+					value:     1,
+				},
+				{
+					chainID:   "1",
+					blockNum:  101,
+					blockHash: "0x0000000000000000000000000000000000000000000000000000000000000456",
+					status:    "future",
+					value:     1,
+				},
+			},
+			expectedInitiatingCalls: []expectedInitiatingCall{
+				{
+					chainID:  "2",
+					blockNum: 200,
+					status:   "future",
+					value:    1,
+				},
+				{
+					chainID:  "2",
+					blockNum: 201,
+					status:   "future",
+					value:    1,
+				},
+			},
+			expectedTerminalCalls: []expectedTerminalCall{},
+		},
+		{
+			name: "jobs across different chains",
+			updater1Jobs: map[JobID]*Job{
+				"job1": jobForTest(1, 100, "0x123", 2, 200, jobStatusFuture),
+			},
+			updater2Jobs: map[JobID]*Job{
+				"job2": jobForTest(2, 300, "0x456", 3, 400, jobStatusValid),
+			},
+			updater3Jobs: map[JobID]*Job{
+				"job3": jobForTest(3, 500, "0x789", 1, 600, jobStatusInvalid),
+			},
+			expectedExecutingCalls: []expectedExecutingCall{
+				{
+					chainID:   "1",
+					blockNum:  100,
+					blockHash: "0x0000000000000000000000000000000000000000000000000000000000000123",
+					status:    "future",
+					value:     1,
+				},
+				{
+					chainID:   "2",
+					blockNum:  300,
+					blockHash: "0x0000000000000000000000000000000000000000000000000000000000000456",
+					status:    "valid",
+					value:     1,
+				},
+				{
+					chainID:   "3",
+					blockNum:  500,
+					blockHash: "0x0000000000000000000000000000000000000000000000000000000000000789",
+					status:    "invalid",
+					value:     1,
+				},
+			},
+			expectedInitiatingCalls: []expectedInitiatingCall{
+				{
+					chainID:  "2",
+					blockNum: 200,
+					status:   "future",
+					value:    1,
+				},
+				{
+					chainID:  "3",
+					blockNum: 400,
+					status:   "valid",
+					value:    1,
+				},
+				{
+					chainID:  "1",
+					blockNum: 600,
+					status:   "invalid",
+					value:    1,
+				},
+			},
+			expectedTerminalCalls: []expectedTerminalCall{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Setup test dependencies
+			logger := log.New()
+			mockMetrics := &mockMetrics{}
+
+			// Track actual metric calls
+			actualExecutingCalls := []expectedExecutingCall{}
+			actualInitiatingCalls := []expectedInitiatingCall{}
+			actualTerminalCalls := []expectedTerminalCall{}
+
+			// Configure mock metrics to record calls
+			mockMetrics.recordExecutingMessageStatsFn = func(chainID string, blockHeight uint64, blockHash string, status string, value float64) {
+				actualExecutingCalls = append(actualExecutingCalls, expectedExecutingCall{
+					chainID:   chainID,
+					blockNum:  blockHeight,
+					blockHash: blockHash,
+					status:    status,
+					value:     value,
+				})
+			}
+			mockMetrics.recordInitiatingMessageStatsFn = func(chainID string, blockHeight uint64, status string, value float64) {
```

Comment:
Nit: there's a lot of stuff here which doesn't depend on `tt`. I wonder if we can extract it into a helper `setupMockMetrics` just to make this anyonymous function a bit shorter.